### DISCUSSION TITLE
How to get the logs for a workflow run

### DISCUSSION THREAD

**Discussion by thakee-kochasoft** [community/community#35402] [Apps, API and Webhooks] [answered]
Hi,

I was able to trigger a workflow dispatch event with the [API](https://docs.github.com/en/rest/actions/workflows#create-a-workflow-dispatch-event).

```
curl \
  -L \
  -X POST \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer mY-g1tHub-4cc3ssT0k3n" \
  -d '{"ref":"main", "inputs":{"test_input":"test_value"} }' \
  https://api.github.com/repos/my-gh-handle/my-repo-name/actions/workflows/test.yml/dispatches
```
Now, I want to view the logs for the above run. To do that I need the run ID, which is not present in the response of the above API. How can I view the logs for the run that I've made above?

**Comment by LangLangBart** [comment, marked as answer]
`workflow ID` and  `workflow RUN ID` are different, but the ID's look similar.

Your workflow (`thakee-kochasoft/Workflowtest/blob/main/.github/workflows/run.yml`) has the ID **36552300**.
- [GitHub Docs workflow](https://docs.github.com/en/rest/actions/workflows#list-repository-workflows)

```zsh
❯ curl -X GET https://api.github.com/repos/thakee-kochasoft/Workflowtest/actions/workflows
{
  "total_count": 1,
  "workflows": [
    {
      "id": 36552300,
       ...
    }
  ]
}
```

Now each RUN for this workflow has a RUN ID, you have run your workflow 5 times and the last ID is **3203076770** and the workflow was triggered by a `workflow_dispatch`. Good, next get the logs.
- [GitHub Docs workflow RUNS](https://docs.github.com/en/rest/actions/workflow-runs#list-workflow-runs-for-a-workflow)

```zsh
❯ curl -X GET https://api.github.com/repos/thakee-kochasoft/Workflowtest/actions/runs
{
  "total_count": 5,
  "workflow_runs": [
    {
      "id": 3203076770,
      "name": "CI",
      "node_id": "WFR_kwLOIIXNnM6-6xKi",
      "head_branch": "main",
      "head_sha": "4c06ccf6a16f8e32c9abf3b35eab6d080749cf5e",
      "path": ".github/workflows/run.yml",
      "display_title": "CI",
      "run_number": 14,
      "event": "workflow_dispatch",
      "status": "completed",
      ...
```

The logs are packed in a zip file, we can run the following command and tell it to save the file as `> log.zip`

```zsh
# for this command you need to have a token
curl \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer <YOUR-TOKEN>" \
  https://api.github.com/repos/thakee-kochasoft/Workflowtest/actions/runs/3203076770/logs > log.zip
```

![SCR-20221007-mm7-2](https://user-images.githubusercontent.com/92653266/194575628-a41fd71f-d566-4266-8beb-1286cec5c564.png)


## Alternatives
### GitHub cli
It is easier to write commands and it is advisable to use this tool if you love the terminal.  The `--repo` flag can be omitted when executed from the directory.
- [gh installation](https://github.com/cli/cli#installation)

```zsh
# trigger a workflow with gh
gh workflow run run.yml --repo thakee-kochasoft/Workflowtest

# list your wokflow RUNS
❯ gh run list --repo thakee-kochasoft/Workflowtest

STATUS  TITLE  WORKFLOW  BRANCH  EVENT              ID          ELAPSED  AGE
✓       CI     CI        main    workflow_dispatch  3203076770  57s      6h
✓       CI     CI        main    workflow_dispatch  3203069849  41s      6h
✓       CI     CI        main    workflow_dispatch  3198579988  45s      22h
✓       CI     CI        main    workflow_dispatch  3184778790  48s      2d
✓       CI     CI        main    workflow_dispatch  3184776949  44s      2d

# show the logs for the RUN ID 3203076770 without downloading the zip file.
❯ gh run view 3203076770 --repo thakee-kochasoft/Workflowtest --log

test-run        Set up job      2022-10-07T07:42:44.8964552Z Current runner version: '2.297.0'
test-run        Set up job      2022-10-07T07:42:44.9005841Z ##[group]Operating System
test-run        Set up job      2022-10-07T07:42:44.9006411Z Microsoft Windows Server 2022
test-run        Set up job      2022-10-07T07:42:44.9006842Z 10.0.20348
test-run        Set up job      2022-10-07T07:42:44.9007163Z Datacenter
test-run        Set up job      2022-10-07T07:42:44.9007540Z ##[endgroup]
...
```

### GitHub cli + extension
Even easier to see all the logs of all the RUNS, when using an extension called [gh-f](https://github.com/gennaro-tedesco/gh-f). 
```zsh
gh f -r
# see GIF below
```

![them](https://user-images.githubusercontent.com/92653266/194577695-97fbd511-48d8-4ee4-bd92-3920b918af6a.gif)
